src/util/session.js

```diff
@@ -203,33 +203,25 @@ export const useSessions = () => {
   });
 };
 
-export const restoreSession = (session) => {
-  const sessionExpires = localStore.getItem('sessionExpires');
-  // We send a request if sessionExpires == null, partly in case there was a
-  // logout error.
-  if (sessionExpires != null && parseInt(sessionExpires, 10) <= Date.now())
-    return Promise.reject();
+export const restoreSession = (session) =>
   // There is a chance that the user's session will be restored almost
   // immediately before the session expires, such that the session expires
   // before logOutBeforeSessionExpires() logs out the user. However, that case
   // is unlikely, and the worst case should be that the user sees 401 messages.
-  return session.request({ url: '/v1/sessions/restore', alert: false })
+  session.request({ url: '/v1/sessions/restore', alert: false })
     .catch(error => {
-      // The user's session may be removed without the user logging out, for
+      // The user's session may be deleted without the user logging out, for
       // example, if a backup is restored. In that case, the request will result
-      // in a 404. sessionExpires may need to be removed from local storage in
-      // order for the user to log in again.
-      if (sessionExpires != null) {
```

Comment:
Removing this `if` means that we will always remove `sessionExpires` after a 404, not just if `sessionExpires` was set to a future timestamp before the request was sent. However, I don't think that's a problem:

- If somehow there's a login in another tab between when this tab sends the request and when it receives the 404 (due to some race condition), any open tab that is logged in will be logged out after the 404. That may be inconvenient, but that sequence of events should also be rare.
- Even if the tab that logged in is closed before the 404 is received, things should mostly be OK. In that case, `sessionExpires` will not be set even though there is a session cookie. However, Frontend will still be able to coordinate logout among tabs.
- In that case, there would be more of an issue in the tab that receives the 404, because `AccountLogin` would allow a login attempt to proceed. However, that safeguard is there just to prevent a confusing user experience. If SSO is not enabled, the user would see a cryptic error message from Backend. If SSO is enabled, and if another tab is opened after the 404 is received, the user will be immediately logged out after logging in.

Given that there weren't tests about this or comments in the code, I think the reason we checked `sessionExpires` before wasn't to prevent these possibilities, but rather to prevent unneeded `storage` events from being triggered (not that those are a big deal either).

I went back and forth on this one, which you can see in the commit history. But given our current setup, including the fact that `restoreSession()` does need to remove `sessionExpires` in some cases, I can't think of a way to prevent these possibilities entirely. And again, this sort of sequence should be unlikely. I suspect we'll come back to this area soon as part of getodk/central#400. For now, I think it's nice for `sessionExpires` to be used in fewer ways.